saleor/plugins/manager.py

```diff
@@ -797,6 +798,41 @@ def list_payment_gateways(
             )
         return gateways
 
+    def list_shipping_methods(
+        self,
+        checkout: Optional["Checkout"] = None,
+        channel_slug: Optional[str] = None,
+        active_only: bool = True,
+    ) -> List["ExternalShippingMethod"]:
+        channel_slug = checkout.channel.slug if checkout else channel_slug
+        plugins = self.get_plugins(channel_slug=channel_slug, active_only=active_only)
+        shipping_plugins = [
+            plugin
+            for plugin in plugins
+            if "get_shipping_methods" in type(plugin).__dict__
+        ]
+
+        shipping_methods = []
+        for plugin in shipping_plugins:
+            shipping_methods.extend(
+                plugin.get_shipping_methods(checkout=checkout, previous_value=None)
+            )
+        return shipping_methods
+
+    def get_shipping_method(
```

Comment:
I am not conviced that this is a proper way for handling this. 
Assuming that you have 3 external shipping methods provider. To fetch single shipping method, you will have to fetch **all** shipping methods for all shipping providers. 
Instead of calling all shipping methods, we should call a single Plugin/Webhook, to retrieve it.
We have a shipping_method_id, it shoud be parsable and should contain enough info to determine which Plugin/App should run this action.
cc @maarcingebala  what do you think? You're more familiar with sync webhook for payments and I believe the approach should be similar here.